src/pgrepr/src/value/interval.rs

```diff
@@ -46,7 +46,10 @@ impl ToSql for Interval {
         // Our intervals are guaranteed to fit within SQL's min/max intervals,
         // so this is compression is guaranteed to be lossless. For details, see
         // `repr::scalar::datetime::compute_interval`.
-        let days = std::cmp::min(self.0.days() as i128, i32::MAX as i128);
+        let days = std::cmp::max(
+            std::cmp::min(self.0.days() as i128, i32::MAX as i128),
```

Comment:
Ah, in the case of `self.0.days() as i128` , you should use `try_from().expect("[I know this works because...]"`.

For `i32::MIN as i128` and the like, you should use `i128::from(i32::MIN)`.